### DISCUSSION TITLE
Process unrouted path

### DISCUSSION THREAD

**Discussion by devmsb** [wfrest/wfrest#194] [Q&A] [answered]
I need to redirect any unrouted path to /index.html. How can I do it?

**Comment by chanchann** [comment, marked as answer]
You can do it in this way : https://github.com/wfrest/wfrest/blob/f09b2c58a14bc642dcd0cb3bab399d0461839b91/example/01_basic.cc#L47

**Comment by devmsb** [comment]
It looks like set_default_route is a new function. I have upgraded sources to 0.9.3 and catched segfault. Wfrest from 10 may 2023 works fine.

**Comment by devmsb** [comment]
free() Invalid pointer in workflow library src/util/json_parser.c

```
static int __set_json_value(int type, va_list ap, json_value_t *val)
{
	json_value_t *src;
	const char *str;
	int len;

	switch (type)
	{
	case 0:
		src = va_arg(ap, json_value_t *);
		__move_json_value(src, val);
		free(src); // <- Invalid pointer here
		return 0;

```
When I'm trying:

data2.push_back("key1", data1["key"]);

data1, data2 are wfrest::Json

When I use old version it works fine.

**Comment by chanchann** [comment]
> free() Invalid pointer in workflow library src/util/json_parser.c
> 
> 
> 
> ```
> 
> static int __set_json_value(int type, va_list ap, json_value_t *val)
> 
> {
> 
> 	json_value_t *src;
> 
> 	const char *str;
> 
> 	int len;
> 
> 
> 
> 	switch (type)
> 
> 	{
> 
> 	case 0:
> 
> 		src = va_arg(ap, json_value_t *);
> 
> 		__move_json_value(src, val);
> 
> 		free(src); // <- Invalid pointer here
> 
> 		return 0;
> 
> 
> 
> ```
> 
> When I'm trying:
> 
> 
> 
> data2.push_back("key1", data1["key"]);
> 
> 
> 
> data1, data2 are wfrest::Json
> 
> 
> 
> When I use old version it works fine.
> 
> 

Please try latest master code

**Reply by devmsb** [reply]
Now this error is gone. Thank you.
But set_default_route is not working as expected.
```
srv.Static("/", "/web/index.html");
srv.set_default_route("/");

```
GET / -> OK
GET /something/other -> 404

**Comment by chanchann** [comment]
https://github.com/wfrest/wfrest/commit/b8947d59834815e716f705d1d15f0ac4738343bf
Sorry for fixing the issue so late.
